modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -84,9 +89,17 @@ public String getHelp() {
     public void processOpts() {
         super.processOpts();
 
-        if (additionalProperties.containsKey(USE_RX_JAVA)) {
+        if (additionalProperties.containsKey(USE_RX_JAVA) && additionalProperties.containsKey(USE_RX_JAVA2)) {
+            LOGGER.warn("You specified both RxJava versions 1 and 2 but they are mutually exclusive. Defaulting to v2.");
+        } else if (additionalProperties.containsKey(USE_RX_JAVA)) {
             this.setUseRxJava(Boolean.valueOf(additionalProperties.get(USE_RX_JAVA).toString()));
         }
+        if (additionalProperties.containsKey(USE_RX_JAVA2)) {
+            this.setUseRxJava2(Boolean.valueOf(additionalProperties.get(USE_RX_JAVA2).toString()));
+        }
+	if (!useRxJava && !useRxJava2) {
+	    additionalProperties.put(DO_NOT_USE_RX, true);
+	}
```

Comment:
That would not work. You can try it with a config such as 
```
{
  "library": "retrofit2",
  "artifactId": "swagger-locator2-retrofit2-rx",
  "useRxJava2": false,
  "hideGenerationTimestamp": true
}
```
and check to see that the results are incorrect with your proposed one liner. My version is correct, however.

Also, feel free to change tab to space but there are other tabs in the file, e.g. right at line 19, which were done by others.